lib/templatesimplifier.cpp

```diff
@@ -3094,6 +3097,7 @@ void TemplateSimplifier::simplifyTemplates(
             mInstantiatedTemplates.clear();
             mExplicitInstantiationsToDelete.clear();
         }
+        mTemplateNamePos.clear();
```

Comment:
This should move up into the if.  It is only needed for the second pass.

Reply:
I see.